lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,337 @@
+import crypto from 'crypto';
+import EventEmitter from 'events';
+import net from 'net';
+import stream from 'stream';
+
+import { errors } from 'arsenal';
+import kineticlib from 'kineticlib';
+
+const HEADER_SZ = 9;
+
+class Kinetic {
+
+    /**
+     * @constructor
+     * @param {Object.<string>[]} drives - the list of drives
+     * @returns {Kinetic} - to allow for a functional style.
+     */
+    constructor(drives) {
+        this._events = new EventEmitter();
+        this._sockets = [];
+        this._drives = drives;
+        for (let i = 0; i < drives.length; i++) {
+            this._sockets.push(
+                { sock: this._newSocket(),
+                  connectionID: 0,
+                  clusterVersion: 0,
+                  drive: drives[i] });
+        }
+        this._request = false;
+        this._chunkSize = 0;
+        this._chunkTab = [];
+        this._count = 0;
+        this.connect();
+        return this;
+    }
+
+    /**
+     * Connect the sockets to the drives
+     * @return {undefined}
+     */
+    connect() {
+        let index = 0;
+        this._sockets.forEach(socket => {
```

Comment:
He's increasing the index in the callback of the ```connect``` event, it can be different from the index given in the callback of forEach.